sessionctx/variable/varsutil.go

```diff
@@ -625,6 +626,23 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 			return "", nil
 		}
 		return value, ErrWrongValueForVar.GenWithStackByArgs(name, value)
+	case TiDBIsolationReadEngines:
+		engines := strings.Split(value, ",")
+		var formatVal string
+		for i, engine := range engines {
+			if i != 0 {
+				formatVal += ","
+			}
+			switch {
+			case strings.EqualFold(engine, kv.TiKV.Name()):
```

Comment:
should we also trim the spaces around each item and remove the duplicated items? For example, users may set the variable this way:
```sql
set @@ tidb_isolation_read_engines = "tiflash, tikv,  tiflash,    tikv";
```

Reply:
Done.